### PR TITLE
support search in factories and add whoosh as a store backend

### PR DESCRIPTION
**Pull request by abom**
#318 is related.

### CODE REVIEW

File: jumpscale/core/base/store/__init__.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+
+        for name in self.list_all():
+            if cursor_ and name != cursor_:
+                continue
+
+            if limit_ and limit_ >= len(found):
+                break
+
+            data = self.get(name)
+            for name, value in kwargs.items():
```
Comment (grimpy) [inline jumpscale/core/base/store/__init__.py:261]: would be nice if this is imple,mented as  generator

File: jumpscale/core/base/store/whooshfts.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
+from whoosh import fields
```
Comment (grimpy) [inline jumpscale/core/base/store/whooshfts.py:1]: what's the idea behind the filename? I don't undersdtand what fts means

Comment (abom) [reply jumpscale/core/base/store/whooshfts.py:1]: "fts" stands for full-text search :laughing: 

I used this not to conflict with the original `whoosh` module.

**Comment by xmonader** [comment]
@abom if good to go please resolve the conflict and merge

**Pull request merged**